File: sdks/ruby/lib/novabilling/types/credit_note_response_reason.rb
# frozen_string_literal: true

module Novabilling
  module Types
    module CreditNoteResponseReason
      extend Novabilling::Internal::Types::Enum

      DUPLICATE = "DUPLICATE"
      PRODUCT_UNSATISFACTORY = "PRODUCT_UNSATISFACTORY"
      ORDER_CHANGE = "ORDER_CHANGE"
      OTHER = "OTHER"
    end
  end
end

File: sdks/rust/src/api/types/credit_note_response_reason.rs
pub use crate::prelude::*;

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
pub enum CreditNoteResponseReason {
    #[serde(rename = "DUPLICATE")]
    Duplicate,
    #[serde(rename = "PRODUCT_UNSATISFACTORY")]
    ProductUnsatisfactory,
    #[serde(rename = "ORDER_CHANGE")]
    OrderChange,
    #[serde(rename = "OTHER")]
    Other,
}
impl fmt::Display for CreditNoteResponseReason {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        let s = match self {
            Self::Duplicate => "DUPLICATE",
            Self::ProductUnsatisfactory => "PRODUCT_UNSATISFACTORY",
            Self::OrderChange => "ORDER_CHANGE",
            Self::Other => "OTHER",
        };
        write!(f, "{}", s)
    }
}
